ethpm_types/abi.py

```diff
@@ -277,6 +279,19 @@ def signature(self) -> str:
 
         return f"{self.name}({input_args}){output_args}"
 
+    @classmethod
+    def from_signature(cls, sig: str) -> Self:
```

Comment:
question: could this also work in `StructABI`?

Reply:
Can't say I've ever used struct signatures before.  Do you think it would be useful?

Reply:
I don't have a use-case in mind! But structurally similar to EventABI so it could be an easy add. Consider it optional!

Reply:
I started playing with the idea, but structs seem way more complex than events.  Just pulling some form the OZ repo showed they can be more complex than I expected.  For instance, they can have mappings and even functions in them.  

```solidity
    struct Bytes32PushTree {
        uint256 _nextLeafIndex;
        bytes32[] _sides;
        bytes32[] _zeros;
        function(bytes32, bytes32) view returns (bytes32) _fnHash;
    }
```

This is deranged!  :laughing: 

For the sake of time, I'm going to punt on this and maybe if someone has a need we can re-evaluate.

Reply:
I think that the actual ABI is a little less crazy looking to me (I think that function pointer ends up as bytes24 or some shit), but it could be very useful to be able to deserialize more complex method signatures

Reply:
I didn't know you could have functions in your structs. Can you call those in Ape??

Reply:
I don't know, just speculating.  But I would bet the pointer @fubuloubu mentioned is just another selector one could call.